tests/gdx-tests/src/com/badlogic/gdx/tests/g3d/shadows/system/BaseShadowSystem.java

```diff
@@ -0,0 +1,468 @@
+	public static class PointLightProperties {
+		public ObjectMap<CubemapSide, LightProperties> properties = new ObjectMap<CubemapSide, LightProperties>(6);
+	}
+
+	/** First pass number */
+	public static final int PASS_1 = 0;
+
+	/** Main camera */
+	protected Camera camera;
+	/** Renderable providers */
+	protected Iterable<RenderableProvider> renderableProviders;
+	/** Cameras linked with spot lights */
+	protected ObjectMap<SpotLight, LightProperties> spotCameras = new ObjectMap<SpotLight, LightProperties>();
+	/** Cameras linked with directional lights */
+	protected ObjectMap<DirectionalLight, LightProperties> dirCameras = new ObjectMap<DirectionalLight, LightProperties>();
+	/** Cameras linked with point lights */
+	protected ObjectMap<PointLight, PointLightProperties> pointCameras = new ObjectMap<PointLight, PointLightProperties>();
+	/** Analyzer of near and far for spot and point lights */
+	protected NearFarAnalyzer nearFarAnalyzer;
+	/** Allocator which choose where to render shadow map in texture */
+	protected ShadowMapAllocator allocator;
+	/** Analyzer which compute how to create the camera for directional light */
+	protected DirectionalAnalyzer directionalAnalyzer;
+	/** Filter that choose if light must be rendered */
+	protected LightFilter lightFilter;
+	/** Framebuffer used to render all the depth maps */
+	protected FrameBuffer[] frameBuffers;
+	/** Current pass in the depth process */
+	protected int currentPass = -1;
+	/** Iterators for cameras */
+	protected Entries<SpotLight, LightProperties> spotCameraIterator;
+	protected Entries<DirectionalLight, LightProperties> dirCameraIterator;
+	protected Entries<PointLight, PointLightProperties> pointCameraIterator;
+	/** Current side in the point light cubemap */
+	protected int currentPointSide;
+	protected PointLightProperties currentPointProperties;
+	/** Tmp variables */
+	protected Vector3 tmpV3 = new Vector3();
+	protected Vector2 tmpV2 = new Vector2();
+	/** Shader providers used by this system */
+	protected ShaderProvider[] passShaderProviders;
+	protected ShaderProvider mainShaderProvider;
+	/** Current light and properties during shadowmap generation */
+	protected LightProperties currentLightProperties;
+	protected BaseLight currentLight;
+
+	/** Construct the system with the needed params.
+	 * @param nearFarAnalyzer Analyzer of near and far
+	 * @param allocator Allocator of shadow maps
+	 * @param directionalAnalyzer Analyze directional light to create orthographic camera
+	 * @param lightFilter Filter light to render */
+	public BaseShadowSystem (NearFarAnalyzer nearFarAnalyzer, ShadowMapAllocator allocator,
+		DirectionalAnalyzer directionalAnalyzer, LightFilter lightFilter) {
+		this.nearFarAnalyzer = nearFarAnalyzer;
+		this.allocator = allocator;
+		this.directionalAnalyzer = directionalAnalyzer;
+		this.lightFilter = lightFilter;
+		frameBuffers = new FrameBuffer[getPassQuantity()];
+		passShaderProviders = new ShaderProvider[getPassQuantity()];
+		init();
```

Comment:
Calling a protected method in the constructor of the base class is not a good idea ;)